src/acquisition/covidcast/database.py

```diff
@@ -213,10 +213,17 @@ def insert_or_update_batch(self, cc_rows, batch_size=2**20, commit_partial=False
 
 
         result = self._cursor.executemany(insert_into_tmp_sql, args)
+
+        # use the temp table to compute meta just for new data, 
+        # then merge with the global meta
+        meta_update = self.compute_covidcast_meta(table_name=tmp_table_name)
```

Comment:
If I understand correctly the (temp) metadata will be updated on every write to the database, is that right? How many times do we expect that to happen each day? (i.e., is it for each signal/source update, or do those happen all at once)

Would a possible alternative be to scan through the database once each day and update then?

Reply:
its pretty much one update to the cache for each input file (csv).  i *believe* the importer runs every so often as a cron job or whatever (multiple times daily?), and loads all new files in the data dump directory.  i also *believe* the files do not arrive on a strict schedule.  one file is unique in terms of (source, signal, time_type (day|week), geo_type (state|county|metro), time_value (yyyymmdd), issue_value (yyyymmdd)).

we could do it once a day but that includes some logic to identify which subset of data should be separated out.  doing it per file gives us the aggregation we need for free.

Reply:
`csv_to_database.py` runs at :50 after every hour, with a no-op exit if there are no files in receiving. So, up to 24x a day, but the true frequency depends on how many of those periods include data drops.

The number of files handled during each run varies from 0 to 53k. About half of the 800 runs in the last log file handled 0 files, 3/4 under 1500 files, but there is that long tail where we'd be merging metadata 10k times in a single run. 

It _might_ be performant as-is, but we should probably think about whether there's a way to efficiently merge a whole pile of these.

Reply:
@melange396 makes sense that this is simpler.

@krivard thanks, so this means we will need to monitor for increased runtime of csv_to_database once this is enabled. 

wdyt about just enabling and monitoring for a day? We can scrape that data as we are doing for the metadata update. Probably best to do it on monday when we are all around.

Reply:
good idea. probably two ways to do it: on staging (easy, but won't see as many data files) or in production (either need to merge & back out a PR, or coordinate with @korlaxxalrok to switch prod to sync to a different branch and then switch back)